src/Lecture4.hs

```diff
@@ -197,7 +243,8 @@ row in the file.
 -}
 
 rowToStats :: Row -> Stats
-rowToStats = error "TODO"
+rowToStats (Row pName Buy pCost) = Stats (Sum 1) (Sum (-pCost)) (Max pCost) (Min pCost) Nothing Nothing (Just $ Max pCost) (Just $ Min pCost) (MaxLen pName)
+rowToStats (Row pName Sell pCost) = Stats (Sum 1) (Sum pCost) (Max pCost) (Min pCost) (Just $ Max pCost) (Just $ Min pCost) Nothing Nothing (MaxLen pName)
```

Comment:
I see that both here and in the `Semigroup` instance for `Stats` you use position arguments for constructors instead of named fields using record syntax. Usually, it's better to use named constructors, especially when some adjacent fields have the same types. It also improves readability 📚